src/urlhandlers/xhr_url_handler.js

```diff
@@ -1,9 +1,13 @@
 export class XHRURLHandler {
   xhr() {
-    const xhr = new window.XMLHttpRequest();
-    if ('withCredentials' in xhr) {
-      // check CORS support
-      return xhr;
+    try {
+      const xhr = new window.XMLHttpRequest();
+      if ('withCredentials' in xhr) {
+        // check CORS support
+        return xhr;
+      }
+    } catch (err) {
+      return false;
```

Comment:
maybe this should `return null` and log an error? Not sure what exactly to do here but returning boolean doesn't seem great to me. In which case can we reach `catch` here?

Reply:
Yes right 👍 

We reach the `catch` when we try to do `new window.XMLHttpRequest();` in an environment where `window` is not defined